hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -1287,6 +1293,8 @@ namespace hazelcast {
                 }
 
                 void SSLSocket::post_connect() {
+                    socket_.set_verify_mode(
+                            boost::asio::ssl::verify_peer | boost::asio::ssl::verify_fail_if_no_peer_cert);
```

Comment:
what if the user explicitly sets the verify mode in the SSL context via https://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/ssl__context/set_verify_mode.html? 

I think we should respect it

Reply:
Also, it says [here](https://www.openssl.org/docs/man1.0.2/man3/SSL_CTX_set_verify.html) that `SSL_VERIFY_FAIL_IF_NO_PEER_CERT` is ignored on the client side. Does it have any use here?

Reply:
Good point. Changed it to respect user config. 
(I initially thought that it was a good idea to enforce but after a second thought I changed my mind on that.)